### PR TITLE
Mkirk/iphone x

### PR DESCRIPTION
**Pull request by michaelkirk**
Fixes a handful of issues around iPhoneX layout.

There should be no noticeable difference for other devices.

I'll attach some screenshots in-line with their corresponding diff.

### CODE REVIEW

File: Signal/src/ViewControllers/CallViewController.swift [hunk 1]
```diff
@@ -569,19 +572,29 @@ class CallViewController: OWSViewController, CallObserver, CallServiceObserver {
...
-
-            contactAvatarView.autoPinEdge(.top, to:.bottom, of:callStatusLabel, withOffset:+avatarTopSpacing)
-            contactAvatarView.autoPinEdge(.bottom, to:.top, of:ongoingCallView, withOffset:-avatarBottomSpacing)
-            contactAvatarView.autoHCenterInSuperview()
-            // Stretch that avatar to fill the available space.
-            contactAvatarView.setContentHuggingLow()
-            contactAvatarView.setCompressionResistanceLow()
-            // Preserve square aspect ratio of contact avatar.
-            contactAvatarView.autoMatch(.width, to:.height, of:contactAvatarView)
+            callStatusLabel.setCompressionResistanceHigh()
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/CallViewController.swift]: **Before**

screen shot 2017-11-07 at 6 07 47 pm

**After**
screen shot 2017-11-08 at 8 49 01 am

**Regression Check**
screen shot 2017-11-08 at 8 49 33 am

File: Signal/src/ViewControllers/FingerprintViewScanController.m [hunk 2]
```diff
@@ -82,7 +82,7 @@ - (void)createViews
     footer.backgroundColor = darkGrey;
     [self.view addSubview:footer];
     [footer autoPinWidthToSuperview];
-    [footer autoPinToBottomLayoutGuideOfViewController:self withInset:0];
+    [footer autoPinEdgeToSuperviewEdge:ALEdgeBottom];
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/FingerprintViewScanController.m]: Please ignore the fact that these screenshots were taken "mid-transition" (hence the previous view being visible on the left edge and the navbar having content in transitition.)

**Before**

screen shot 2017-11-08 at 9 11 24 am

**After**
screen shot 2017-11-08 at 8 52 48 am

**Regression Check**
screen shot 2017-11-08 at 8 52 01 am

File: Signal/src/ViewControllers/OWSViewController.m [hunk 3]
```diff
@@ -110,7 +110,8 @@ - (void)handleKeyboardNotification:(NSNotification *)notification
     CGRect keyboardEndFrameConverted = [self.view convertRect:keyboardEndFrame fromView:nil];
     // Adjust the position of the bottom view to account for the keyboard's
     // intrusion into the view.
-    CGFloat offset = -MAX(0, (self.view.height - keyboardEndFrameConverted.origin.y));
+    // We offset by the bottomLayoutGuide for iPhoneX users, else there is an unnecessary gap between the keyboard and input bar.
+    CGFloat offset = -MAX(0, (self.view.height - keyboardEndFrameConverted.origin.y - self.bottomLayoutGuide.length));
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/OWSViewController.m]: This is the one I was the least sure of. Take an extra look @charlesmchen.

**Before**
screen shot 2017-11-08 at 9 10 51 am

**After**
screen shot 2017-11-08 at 9 15 21 am


**Regression Check**
screen shot 2017-11-08 at 9 14 53 am

File: Signal/src/ViewControllers/NewContactThreadViewController.m [hunk 4]
```diff
@@ -75,6 +75,7 @@ - (void)loadView
 {
     [super loadView];
 
+    self.view.backgroundColor = UIColor.whiteColor;
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/NewContactThreadViewController.m]: **Before**

screen shot 2017-11-08 at 9 13 02 am

**After**

image

**Regression Check**
screen shot 2017-11-08 at 9 13 34 am

File: Signal/src/ViewControllers/RegistrationViewController.m [hunk 5]
```diff
@@ -61,41 +61,46 @@ - (void)viewDidLoad {
...
     self.view.userInteractionEnabled = YES;
     [self.view
         addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(backgroundTapped:)]];
 
-    UIView *header = [UIView new];
-    header.backgroundColor = [UIColor ows_signalBrandBlueColor];
-    [self.view addSubview:header];
-    [header autoPinToTopLayoutGuideOfViewController:self withInset:0];
-    [header autoPinWidthToSuperview];
+    UIView *headerWrapper = [UIView containerView];
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/RegistrationViewController.m]: **Before**
screen shot 2017-11-07 at 5 33 10 pm

**After**
screen shot 2017-11-07 at 5 32 23 pm

**Regression Check**
screen shot 2017-11-08 at 9 20 57 am

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 6]
```diff
@@ -479,6 +479,13 @@ - (void)viewDidLoad
     [self loadDraftInCompose];
 }
 
+- (void)loadView
+{
+    [super loadView];
+
+    self.view.backgroundColor = [UIColor ows_inputToolbarBackgroundColor];
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: **Before**
screen shot 2017-11-08 at 9 11 16 am

**After**
screen shot 2017-11-08 at 9 15 18 am

**Regression Check**
image

File: Signal/src/ViewControllers/CallViewController.swift [hunk 7]
```diff
@@ -569,19 +572,29 @@ class CallViewController: OWSViewController, CallObserver, CallServiceObserver {
...
+            // Ensure ContacAvatarView gets as close as possible to it's superview edges while maintaining
+            // aspect ratio.
+            contactAvatarView.autoPinToSquareAspectRatio()
+            contactAvatarView.autoPinEdge(toSuperviewEdge: .top, withInset: 0, relation: .greaterThanOrEqual)
+            contactAvatarView.autoPinEdge(toSuperviewEdge: .right, withInset: 0, relation: .greaterThanOrEqual)
+            contactAvatarView.autoPinEdge(toSuperviewEdge: .bottom, withInset: 0, relation: .greaterThanOrEqual)
+            contactAvatarView.autoPinEdge(toSuperviewEdge: .left, withInset: 0, relation: .greaterThanOrEqual)
+            NSLayoutConstraint.autoSetPriority(UILayoutPriorityDefaultLow) {
+                contactAvatarView.autoPinEdgesToSuperviewMargins()
+            }
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/CallViewController.swift]: Sweet. I didn't know about `autoSetPriority`

**Comment by michaelkirk** [comment]
FYI @charlesmchen  I snuck one more commit in before merging to fix the gif picker footer: https://github.com/WhisperSystems/Signal-iOS/pull/2740/commits/a27b0340902f67bb979e21ba8aa383eaa0b54b8b

**Pull request merged**